org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JobHelpers.java

```diff
@@ -226,6 +234,23 @@ public static void waitForProjectRegistryRefreshJob() {
 		waitForJobs(ProjectRegistryRefreshJobMatcher.INSTANCE, MAX_TIME_MILLIS);
 	}
 
+	// copied from ./org.eclipse.jdt.core.tests.performance/src/org/eclipse/jdt/core/tests/performance/FullSourceWorkspaceTests.java
```

Comment:
You can also find it in https://github.com/eclipse-jdt/eclipse.jdt.core/blob/master/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/AbstractJavaModelTests.java#L4003 . I'm pretty sure this is the right thing to do as I remember using it in the past to wait. :+1: